src/main/java/com/devathon/preguntonic/controllers/RoomControllerImpl.java

```diff
@@ -65,79 +74,72 @@ public ResponseEntity<Room> getRoom(final String roomId) {
   @Override
   public ResponseEntity<LobbyEvent> joinRoom(final String roomId, final BasicPlayer player) {
     log.info("Joining room {} with user {}", roomId, player.name());
-    Optional<Room> room = roomService.getRoom(roomId);
-    if (room.isEmpty()) {
-      log.info(ROOM_NOT_FOUND, roomId);
-      return ResponseEntity.notFound().build();
-    }
-    int playerId = -1;
-    // Special cases for the room creator!
-    if (Objects.nonNull(player.id())) {
-      playerId = player.id();
-    } else {
-      playerId = roomService.joinRoom(roomId, player.name());
+    try {
+      BasicPlayer playerResponse = roomService.joinRoom(roomId, player);
+      log.info("Player {} joined room {}", playerResponse.id(), roomId);
+    } catch (final InvalidParameterException e) {
+      log.warn(ERROR_JOINING_ROOM_MSG, e);
+      return ResponseEntity.badRequest().build();
+    } catch (final Exception e) {
+      log.warn(ERROR_JOINING_ROOM_MSG, e);
+      return ResponseEntity.internalServerError().build();
     }
 
-    var lobbyEvent =
-        LobbyEvent.builder()
-            .event(RoomEvent.JOIN)
-            .roomStatus(LobbyStatusDto.from(room.get()))
-            .build();
-
-    messagingTemplate.convertAndSend(ROOM_DESTINATION_BASE_PATH + roomId, lobbyEvent);
-
-    log.info("Player {} joined room {}", playerId, roomId);
+    var lobbyEvent = sentLobbyEventToRoomTopic(roomId, RoomEvent.JOIN);
     return ResponseEntity.ok(lobbyEvent);
   }
 
   @Override
   public ResponseEntity<LobbyEvent> playerReadyInRoom(final String roomId, final int playerId) {
     log.info("Player {} is ready in room {}", playerId, roomId);
-    Optional<Room> room = roomService.getRoom(roomId);
-    if (room.isEmpty()) {
-      log.info(ROOM_NOT_FOUND, roomId);
-      return ResponseEntity.notFound().build();
+    try {
+      roomService.changePlayerReadyStatus(roomId, playerId, true);
+    } catch (final InvalidParameterException e) {
+      log.warn(PLAYER_READY_ERROR_MSG, e);
+      return ResponseEntity.badRequest().build();
+    } catch (final Exception e) {
+      log.warn(PLAYER_READY_ERROR_MSG, e);
+      return ResponseEntity.internalServerError().build();
```

Comment:
WebSocket works with ResponseEntity?

Reply:
Really we can return anything I decide use this to have a wrapper and a well defined type of responses.